cli/src/nonce.rs

```diff
@@ -0,0 +1,466 @@
+                        .validator(is_pubkey_or_keypair)
+                        .help("Address of the nonce account to display"),
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("withdraw-from-nonce-account")
+                .about("Withdraw lamports from the nonce account")
+                .arg(
+                    Arg::with_name("nonce_account_keypair")
+                        .index(1)
+                        .value_name("NONCE ACCOUNT")
+                        .takes_value(true)
+                        .required(true)
+                        .validator(is_keypair_or_ask_keyword)
+                        .help("Nonce account from to withdraw from"),
+                )
+                .arg(
+                    Arg::with_name("destination_account_pubkey")
+                        .index(2)
+                        .value_name("DESTINATION ACCOUNT")
+                        .takes_value(true)
+                        .required(true)
+                        .validator(is_pubkey_or_keypair)
+                        .help("The account to which the lamports should be transferred"),
+                )
+                .arg(
+                    Arg::with_name("amount")
+                        .index(3)
+                        .value_name("AMOUNT")
+                        .takes_value(true)
+                        .required(true)
+                        .help("The amount to withdraw from the nonce account (default unit SOL)"),
+                )
+                .arg(
+                    Arg::with_name("unit")
+                        .index(4)
+                        .value_name("UNIT")
+                        .takes_value(true)
+                        .possible_values(&["SOL", "lamports"])
+                        .help("Specify unit to use for request"),
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("show-nonce-account")
+                .about("Show the contents of a nonce account")
+                .arg(
+                    Arg::with_name("nonce_account_pubkey")
+                        .index(1)
+                        .value_name("NONCE ACCOUNT")
+                        .takes_value(true)
+                        .required(true)
+                        .validator(is_pubkey_or_keypair)
+                        .help("Address of the nonce account to display"),
+                )
+                .arg(
+                    Arg::with_name("lamports")
+                        .long("lamports")
+                        .takes_value(false)
+                        .help("Display balance in lamports instead of SOL"),
+                ),
+        )
+    }
+}
+
+pub fn parse_nonce_create_account(matches: &ArgMatches<'_>) -> Result<CliCommandInfo, CliError> {
+    let nonce_account = keypair_of(matches, "nonce_account_keypair").unwrap();
+    let lamports = amount_of(matches, "amount", "unit").expect("Invalid amount");
```

Comment:
I know we do this expect a few places in cli.rs and stake.rs, but can we `.ok_or()` to return an error here instead? 

Alternately, we could add a validator to the `lamports` Args to check they are parseable as ints, making this panic unreachable.

(I'll fixup the non-Nonce spots to follow your lead :) )

Reply:
I added a validator :-)